x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/Aggregate.java

```diff
@@ -15,29 +18,87 @@
 import org.elasticsearch.xpack.esql.core.plan.logical.UnaryPlan;
 import org.elasticsearch.xpack.esql.core.tree.NodeInfo;
 import org.elasticsearch.xpack.esql.core.tree.Source;
+import org.elasticsearch.xpack.esql.io.stream.PlanStreamInput;
+import org.elasticsearch.xpack.esql.io.stream.PlanStreamOutput;
 
+import java.io.IOException;
 import java.util.List;
 import java.util.Objects;
 
+import static org.elasticsearch.xpack.esql.io.stream.PlanNameRegistry.PlanReader.readerFromPlanReader;
+import static org.elasticsearch.xpack.esql.io.stream.PlanNameRegistry.PlanWriter.writerFromPlanWriter;
+
 public class Aggregate extends UnaryPlan {
 
+    public enum AggregateType {
+        STANDARD,
+        // include metrics aggregates such as rates
+        METRICS;
```

Comment:
Nit: METRICS_ONLY? I think the standard ones apply to metrics, e.g. avg and max.

Reply:
The METRICS aggregate type might include rate aggregates, but the STANDARD type cannot include rate aggregates. Are you okay with these enum names?

Reply:
This is fine, I'd add a comment to STANDARD to denote that it describes aggregates that apply to all types, including metrics, whereas METRICS include aggs applicable to time series only.